test/parsing/precedence/bang-vs-new-etc.good

```diff
@@ -12,6 +12,24 @@ bang-vs-new-etc.chpl:135: warning: z18b = xx : class? : unmanaged C?  expected:
 bang-vs-new-etc.chpl:143: warning: z19b = new C()? : unmanaged  expected: unmanaged C?  actual: unmanaged C?
 bang-vs-new-etc.chpl:146: warning: z19c = new C?() :unmanaged  expected: unmanaged C?  actual: unmanaged C?
 bang-vs-new-etc.chpl:151: warning: z19d = new C(): unmanaged class?  expected: unmanaged C?  actual: unmanaged C?
+bang-vs-new-etc.chpl:161: warning: unnecessary type construction call
+bang-vs-new-etc.chpl:6: note: 'C' has no generic fields
+bang-vs-new-etc.chpl:161: note: remove the ()
+bang-vs-new-etc.chpl:167: warning: unnecessary type construction call
+bang-vs-new-etc.chpl:6: note: 'C' has no generic fields
+bang-vs-new-etc.chpl:167: note: remove the ()
+bang-vs-new-etc.chpl:170: warning: unnecessary type construction call
+bang-vs-new-etc.chpl:6: note: 'C' has no generic fields
+bang-vs-new-etc.chpl:170: note: remove the ()
+bang-vs-new-etc.chpl:174: warning: unnecessary type construction call
+bang-vs-new-etc.chpl:6: note: 'C' has no generic fields
+bang-vs-new-etc.chpl:174: note: remove the ()
+bang-vs-new-etc.chpl:177: warning: unnecessary type construction call
+bang-vs-new-etc.chpl:6: note: 'C' has no generic fields
+bang-vs-new-etc.chpl:177: note: remove the ()
+bang-vs-new-etc.chpl:180: warning: unnecessary type construction call
+bang-vs-new-etc.chpl:6: note: 'C' has no generic fields
+bang-vs-new-etc.chpl:180: note: remove the ()
```

Comment:
These error messages don't seem to improve the situation to me.  What we should be encouraging is surrounding the `new C()` portion with `()`, I think suggesting the opposite is just confusing

Reply:
Actually, I'm not sure what I can do about this one.

In the test, `new C().xmeth(5.5);`, `xmeth` is *only* a type method. I don't think the way this test is written gives me any way to know that the author intended to write `(new C()).method`. Since this test is written to check the parser precedence levels for various operations, and isn't reflective of patterns users are likely to write, I'd like to simply put the warnings in the .good file and not worry about it further.

Reply:
Hmm, I think I want to think about this a bit longer

Reply:
> What we should be encouraging is surrounding the new C() portion with ()

Lydia, I read it that way at first too, but isn't what we'd want to encourage writing actually more like:

```chapel
  var z20a = new (C.tmeth)();
```

The use of the paren-less method and its choice of name are arguably the real pain points here and arguably bad style.  E.g., `new C.eltType();` seems like it would less ambiguously be creating an instance of a type's type field, and/or `new C.getEltType()();` would be less ambiguously calling a type method and then creating an instance of it (not that I'm super-wild about the latter form either, and would probably either parenthesize more or capture the type method into a type alias before invoking `new` on it).

In any case, given that we're calling a type method and `C` is generic, I'd agree that `C.` is clearer than `C().` and think this is a step forward (though I'd probably update the test rather than the .good file to improve clarity; and add another separate test of the warning in a case like this).

Reply:
BTW, in an off-github discussion, Lydia and I agreed that her concern is a bit of a different case and would be best met by creating a new .future.

>  (though I'd probably update the test rather than the .good file to improve clarity; and add another separate test of the warning in a case like this).

I'll update this test to make C generic-with-defaults to avoid the warning.

Reply:
> I'll update this test to make C generic-with-defaults to avoid the warning.

Hmm, that did not work. I'll clone the parts of the test generating the warning & comment them out in the main test.